### PR TITLE
Add markdown editor

### LINKED ISSUE #28: Adding optional markdown editor

### PR DESCRIPTION
**Pull request by AbhayVAshokan**
Fixes #28 
- Added markdown editor.
- Updated documentation.
- Uses [marked](https://www.npmjs.com/package/marked) package for markdown to HTML conversion.
- Uses [turndown](https://www.npmjs.com/package/turndown) package for HTML to markdown conversion.

Please go through this video: https://vimeo.com/656211314/5713fc4066

@amaldinesh7 _a please have a look.

### CODE REVIEW

**Comment by AbhayVAshokan** [comment]
@amaldinesh7 _a the PR is ready for review.
New changes: https://vimeo.com/658620411/e5e9c7dc92

**Comment by amaldinesh7** [comment]
@labeebklatif _a Please take a look at this PR

**Comment by AbhayVAshokan** [comment]
Rebased with master to resolve conflicts, it automatically fixed the minor slash commands bug mentioned in the video.

**Comment by labeebklatif** [comment]
@AbhayVAshokan Please have a look at this: https://www.loom.com/share/fdf1461408924dc59ba56feb007857ab

**Comment by labeebklatif** [comment]
@AbhayVAshokan _a assign back once the issues in the video are addressed

**Comment by yedhink** [comment]
@AbhayVAshokan About adding line break in markdown:
- https://markdown-guide.readthedocs.io/en/latest/basics.html#line-return
- https://stackoverflow.com/questions/33191744/how-to-add-new-line-in-markdown-presentation

cc: @labeebklatif

**Comment by labeebklatif** [comment]
> @AbhayVAshokan About adding line break in markdown:
> 
> * https://markdown-guide.readthedocs.io/en/latest/basics.html#line-return
> * https://stackoverflow.com/questions/33191744/how-to-add-new-line-in-markdown-presentation
> 
> cc: @labeebklatif

@yedhink Right. We realized markdown needs special syntax to indicate line breaks. But still, the issue exists when we add line breaks in Rich Text mode which is not reflected in the markdown because the line breaks are indicated as empty `p` tags rather than `br` tag in the editor content. @AbhayVAshokan is looking in to this.

File: lib/components/Editor/CustomExtensions/Markdown/TabSwitcher.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import Tab from "components/Common/Tab";
+
+const MarkdownTabSwitcher = ({ isMarkdownTab, setIsMarkdownTab }) => {
```
Comment (labeebklatif) [inline lib/components/Editor/CustomExtensions/Markdown/TabSwitcher.js:4]:
```suggestion
const MarkdownTabSwitcher = ({ isMarkdownTabActive, setIsMarkdownTabActive }) => {
```

File: lib/components/Editor/index.js [hunk 2]
```diff
@@ -29,6 +32,7 @@ const Tiptap = (
   ref
 ) => {
   const [isImageUploadVisible, setImageUploadVisible] = useState(false);
+  const [isMarkdownTab, setIsMarkdownTab] = useState(false);
```
Comment (labeebklatif) [inline lib/components/Editor/index.js:35]:
```suggestion
  const [isMarkdownTabActive, setIsMarkdownTabActive] = useState(false);
```

File: lib/components/Editor/CustomExtensions/Markdown/index.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  useEffect(() => {
+    return () => editor.commands.setContent(markdownToHtml(contentRef.current));
+  }, []);
+
+  useLayoutEffect(() => {
+    const BORDER_HEIGHT = 1;
+
+    textareaRef.current.style.height = "inherit";
+    textareaRef.current.style.height =
+      textareaRef.current.scrollHeight + 2 * BORDER_HEIGHT + "px";
```
Comment (labeebklatif) [inline lib/components/Editor/CustomExtensions/Markdown/index.js:27]: Please check if this auto adjusting height can be achieved by css min-height as we have done for the Rich Text editor part.

**Review by labeebklatif — CHANGES_REQUESTED** [review]
@AbhayVAshokan _a Added some comments. Please go through them.

**Comment by AbhayVAshokan** [comment]
@labeebklatif _a take a look at the changes.
- Fixed the link dropdown issue in #107, rebased with the latest master.
- Resolved PR reviews.
- Added logic to persist new lines in rich text mode.
Markdown linebreak

**Review by labeebklatif — APPROVED** [review]
LGTM!!

**Pull request merged**